plaso/storage/redis/writer.py

```diff
@@ -0,0 +1,357 @@
+# -*- coding: utf-8 -*-
+"""Storage writer for Redis."""
+
+from __future__ import unicode_literals
+
+from plaso.lib import definitions
+from plaso.storage import interface
+from plaso.storage.redis import redis_store
+
+
+class RedisStorageWriter(interface.StorageWriter):
+  """Redis-based storage writer."""
+
+  def __init__(
+      self, session, storage_type=definitions.STORAGE_TYPE_TASK, task=None):
+    """Initializes a storage writer.
+
+    Args:
+      session (Session): session the storage changes are part of.
+      storage_type (Optional[str]): storage type.
+      task(Optional[Task]): task.
+
+    Raises:
+      RuntimeError: if no task is provided.
+    """
+    if not task:
+      raise RuntimeError('Task required.')
+
+    super(RedisStorageWriter, self).__init__(
+        session=session, storage_type=storage_type, task=task)
+
+    self._redis_namespace = '{0:s}-{1:s}'.format(
+        task.session_identifier, task.identifier)
+    self._store = None
+
+  def CreateTaskStorage(self, task, task_storage_format):
+    """Creates a task storage.
+
+    Args:
+      task (Task): task.
+      task_storage_format (str): storage format to store task results.
+
+    Raises:
+      IOError: if the storage type does not support creating a task store or
+          if the task storage already exists.
+      OSError: if the storage type does not support creating a task store or
+          if the task storage already exists.
+    """
+    raise IOError(
+        'Creating a task storage is not supported by the redis store.')
+
+  def FinalizeTaskStorage(self, task):
+    """Finalizes a processed task storage.
+
+    Args:
+      task (Task): task.
+
+    Raises:
+      IOError: if the storage type does not support finalizing a task store.
+      OSError: if the storage type does not support finalizing a task store.
+    """
+    raise IOError(
+        'Finalizing a task storage is not supported by the redis store.')
+
+  # pylint: disable=arguments-differ
+  def Open(self, redis_client=None):
```

Comment:
same suggestion about using unused_kwargs

Reply:
add it anyway, pylint seems to be changing its mind from time to time also see https://github.com/log2timeline/plaso/pull/2941

Reply:
Done